sanity_html/renderer.py

```diff
@@ -23,21 +23,24 @@ def __init__(self, blocks: list[dict]) -> None:
 
     def render(self) -> str:
         """Render HTML from self._blocks."""
-        rendered_html = ''
-        list_nodes: List[Dict] = []
+        result = ''
+        list_nodes: List[dict] = []
         for node in self._blocks:
 
             if list_nodes and not is_list(node):
-                rendered_html += self._render_list(list_nodes)
+                result += self._render_list(list_nodes)
                 list_nodes = []  # reset list_nodes
 
             if is_list(node):
                 list_nodes.append(node)
                 continue  # handle all elements ^ when the list ends
 
-            rendered_html += self._render_node(node)  # render non-list nodes immediately
+            result += self._render_node(node)  # render non-list nodes immediately
 
-        return rendered_html.strip()
+        if list_nodes:
+            result += self._render_list(list_nodes)
```

Comment:
We weren't handling leftover list elements, so if we ended on a list element we were getting bad results.